### PR TITLE
INT-3516, INT-3517 - Pagination and rate limiting update

### PR DESCRIPTION
**Pull request by eXtremeX**
# Description

Thank you for contributing to a JupiterOne integration!

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Summary

- Fixed the pagination in `fetchAgents` step. It was previously checking a non-existent property, `next`, which was not found even in older versions of the API doc.
- Moved the pagination of agents to the `WazuhClient` instead of inside `fetchAgents` step.
- Updated the way tokens are refreshed. It was previously using `setInterval()` and calls an incorrect endpoint that updates the setting for the token expiry and does not update the current token. It is now handled in the `retry()` method.
- Updated the endpoint used in `verifyAccess()` based on the recommendation on the API doc.
- Fixed the checking of URL to ignore domain and port matching for `yarn test`

## Type of change

Please leave any irrelevant options unchecked.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [X] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [X] My changes properly paginate the target service provider's API
- [X] My changes properly handle rate limiting of the target service provider's
      API
- [ ] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [ ] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [X] I have updated the `CHANGELOG.md` file to describe my changes
- [ ] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/wazuh/client.ts [hunk 1]
```diff
@@ -2,26 +2,28 @@ import {
...
   private requestOptions: RequestInit;
-
   private config: WazuhClientConfig;
-
   private refreshAuthInterval: NodeJS.Timer;
-
   private logger: IntegrationLogger;
-
   public initialized: boolean;
+  private pageSize = 500;
```
Comment (mdaum) [inline src/wazuh/client.ts:26]: If we don't intend on changing this and is effectively "class static", then we could move it to top of this file and follow our naming convention for such vars: `const PAGE_SIZE = 500`
Not holding PR up over this though.

File: src/wazuh/client.ts [hunk 2]
```diff
@@ -177,18 +161,79 @@ class WazuhClient {
...
+            });
+          } else {
+            return res;
+          }
+        },
+        {
+          delay: 5000,
+          maxAttempts: 10,
+          handleError: async (err, context) => {
+            if (err.statusCode === 401) {
```
Comment (mdaum) [inline src/wazuh/client.ts:217]: if this is now handling expired JWTs, we should add a comment for better readability.

Comment (d10p3t) [reply src/wazuh/client.ts:215]: Added comments, as suggested.

File: src/wazuh/client.ts (hunk 1 above)
Comment (d10p3t) [reply src/wazuh/client.ts]: Changed this to a `const`, as suggested

**Comment by eXtremeX** [comment]
@mdaum Ready for another peek when you're free!

**Review by mdaum — APPROVED** [review]

**Pull request merged**